### ISSUE TITLE
Safe loading does not work, & custom object support

### ISSUE THREAD

**Issue by jordan-melendez** [telegraphic/hickle#150]
## Safe Loading
I noticed that in one of the recent releases, the `safe` key no longer appears to be working inside of `load`. Looking at the source, the argument is only passed to legacy loaders. Was this intentional? If not, can this functionality be resurrected? One of the main draws to this great package is the fact that I can try to avoid arbitrary code executing if I so choose.

Here is an example:
```python
import hickle

class MyCustomClass:
    def __init__(self, a, b):
        self.a = a
        self.b = b
        
    def __repr__(self):
        return f"{self.__class__.__name__}({self.a}, {self.b})"
        
c = MyCustomClass(1, 2)
print(c)
with open("my_class.hkl", "w") as f:
    hickle.dump(c, f)

with open("my_class.hkl", "r") as f:
    c2 = hickle.load(f, safe=True)
print(c2)
```

## Custom Object Support
And maybe I should create a separate issue for this next question, but I was wondering if there is a way to support custom classes without adding them to this package specifically. It would be fantastic if I could do something like

```python
from hickle.lookup import register_class
register_class(
    MyCustomClass,
   "MyCustomClass",
    dump_function,
    load_function,
)
```
and have it just work. Does this functionality exist? If so, could I see an example? If not, how big of an ask is this to add?

**Comment by 1313e** [comment]
The `safe` keyword was removed in `hickle` v4 as it no longer serves any purpose.
In v4, we started using the `literal_eval` function of the `ast` built-in library, which always avoids arbitrary code execution.
Therefore, the `safe` keyword was no longer needed.

Custom object support exists already in `hickle` in exactly the form you give as an example. :)
You can look at the different loaders for ideas: https://github.com/telegraphic/hickle/tree/master/hickle/loaders

**Comment by hernot** [comment]
@jordan-melendez: Are custom classes supported without adding dedicated loader module to hickle package: No&Yes 
**No:** Not in hickle 4.x exempt you are ok with your class being stored as pickle string. 
**Yes**: Unless it is very urgent you might have a look at compact expand protocol as proposed by H4EP003 #145 if that would fullfill your needs. A first implementation is included in the pull-request #149. In case you want to run some tests whether proposed compact-expand protocol it will fit your needs you may download and install a first snapshot from my [hickle repo](https://github.com/hernot/hickle/tree/hickle-5-RC) fork on github. But keep in mind it is a first release candidate which will undergo still plenty of changes until finally considered for release by @1313e  and @telegraphic and therefore has to be considered unstable at the moment. So use at or own risk and do not use in production. Further report any issue directly to pull-request #149, in case you have questions regarding the proposed compact-expand protocol directly comment on issue #145.
